src/main/java/com/sprint/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -39,40 +43,48 @@ public Channel create(PublicChannelCreateRequest request) {
     }
 
     @Override
+    @Transactional
     public Channel create(PrivateChannelCreateRequest request) {
         Channel channel = new Channel(ChannelType.PRIVATE, null, null);
         Channel createdChannel = channelRepository.save(channel);
 
-        request.getParticipantIds().stream()
-                .map(userId -> new ReadStatus(userId, createdChannel.getId(), Instant.MIN))
-                .forEach(readStatusRepository::save);
+        request.getParticipantIds().forEach(userId -> {
+            User user = userRepository.findById(userId).orElse(null);
+            ReadStatus readStatus = ReadStatus.builder()
+                    .user(user)
+                    .channel(channel)
+                    .lastReadAt(createdChannel.getCreatedAt())
+                    .build();
 
+            readStatusRepository.save(readStatus);
+        });
         return createdChannel;
     }
 
     @Override
     public ChannelDto find(UUID channelId) {
         return channelRepository.findById(channelId)
-                .map(this::toDto)
+                .map(channelMapper::toDto)
                 .orElseThrow(() -> new NoSuchElementException("Channel with id " + channelId + " not found"));
     }
 
     @Override
     public List<ChannelDto> findAllByUserId(UUID userId) {
-        List<UUID> mySubscribedChannelIds = readStatusRepository.findAllByUserId(userId).stream()
-                .map(ReadStatus::getChannelId)
+        List<UUID> mySubscribedChannelIds = readStatusRepository.findAllByUser_Id(userId).stream()
```

Comment:
@minhyuksim 메서드 이름에 _(언더바)를 쓰는 건 일반적이지 않습니다. 지워주시는 게 좋아요.

Reply:
@minhyuksim 이건 JPQL로 풀어볼 수도 있을 것 같아요.

```suggestion
        //Repository
        @Query("SELECT c FROM Channel c WHERE c.type='PUBLIC' OR c.id IN :ids")
```